systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java

```diff
@@ -0,0 +1,328 @@
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaConnectUtils;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaTopicUtils;
+import io.strimzi.systemtest.utils.kubeUtils.controllers.DeploymentUtils;
+import io.strimzi.systemtest.utils.kubeUtils.objects.PodUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.strimzi.systemtest.Constants.CONNECT;
+import static io.strimzi.systemtest.Constants.CONNECT_COMPONENTS;
+import static io.strimzi.systemtest.Constants.REGRESSION;
+import static io.strimzi.systemtest.enums.CustomResourceStatus.NotReady;
+import static io.strimzi.systemtest.enums.CustomResourceStatus.Ready;
+import static io.strimzi.test.k8s.KubeClusterResource.cmdKubeClient;
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@Tag(REGRESSION)
+@Tag(CONNECT_COMPONENTS)
+@Tag(CONNECT)
+class ConnectBuilderST extends AbstractST {
+
+    public static final String NAMESPACE = "connect-builder-cluster-test";
+    private static final Logger LOGGER = LogManager.getLogger(ConnectBuilderST.class);
+
+    private static final String ECHO_SINK_CLASS_NAME = "cz.scholz.kafka.connect.echosink.EchoSinkConnector";
+    private static final String CAMEL_TELEGRAM_CLASS_NAME = "org.apache.camel.kafkaconnector.telegram.CamelTelegramSinkConnector";
+
+    private String imageName = "";
+
+    private static final Plugin PLUGIN_WITH_TAR_AND_JAR = new PluginBuilder()
+        .withName("connector-with-tar-and-jar")
+        .withArtifacts(
+            new JarArtifactBuilder()
+                .withNewUrl("https://github.com/scholzj/echo-sink/releases/download/1.1.0/echo-sink-1.1.0.jar")
+                .withNewSha512sum("b7da48d5ecd1e4199886d169ced1bf702ffbdfd704d69e0da97e78ff63c1bcece2f59c2c6c751f9c20be73472b8cb6a31b6fd4f75558c1cb9d96daa9e9e603d2")
+                .build(),
+            new TgzArtifactBuilder()
+                .withNewUrl("https://github.com/scholzj/echo-sink/archive/1.1.0.tar.gz")
+                .withNewSha512sum("5318b1f031d4e5eeab6f8b774c76de297237574fc51d1e81b03a10e0b5d5435a46a108b85fdb604c644529f38830ae83239c17b6ec91c90a60ac790119bb2950")
+                .build())
+        .build();
+
+    @Test
+    void testBuildFailsWithWrongChecksumOfArtifact() {
+        Plugin pluginWithWrongChecksum = new PluginBuilder()
+            .withName("connector-with-empty-checksum")
+            .withArtifacts(new JarArtifactBuilder()
+                .withNewUrl("https://github.com/scholzj/echo-sink/releases/download/1.1.0/echo-sink-1.1.0.jar")
+                .withNewSha512sum("ads")
```

Comment:
I think this test might be more interesting if it uses some proper checksum but not matching the file (For example a checksum of some other random file).